docs/internals/optimizer.rst

```diff
@@ -20,17 +23,23 @@ the function calls. Similarly, if a function is
 side-effect free and its result is multiplied by zero, you can remove the function
 call completely.
 
+The codegen-based optimizer consists of two steps, the first of which changes the
+order of literals in binary operators, moving them to the right. The second one
+enables the compiler to utilize unchecked arithmetic when generating code for
+incrementing the counter variable in certain ``for`` loops.
+
 Currently, the parameter ``--optimize`` activates the opcode-based optimizer for the
 generated bytecode and the Yul optimizer for the Yul code generated internally, for example for ABI coder v2.
 One can use ``solc --ir-optimized --optimize`` to produce an
 optimized Yul IR for a Solidity source. Similarly, one can use ``solc --strict-assembly --optimize``
 for a stand-alone Yul mode.
 
 .. note::
-    The `peephole optimizer <https://en.wikipedia.org/wiki/Peephole_optimization>`_ is always
+    The `peephole optimizer <https://en.wikipedia.org/wiki/Peephole_optimization>`_ and
+    the :ref:`unchecked loop increment optimizer <unchecked-loop-optimizer>` are always
     enabled by default and can only be turned off via the :ref:`Standard JSON <compiler-api>`.
```

Comment:
Oh, one more thing I just realized while looking at the same spot in #14657. This originally mentioned only the peephole optimizer but there are more steps that are on by default (e.g. the jumpdest remover).

I'm sure we will not be able to keep it in sync once we add more steps like this. Maybe it would be best to just make it list those steps as examples, and not make it seem like it's an exhaustive list. I.e. say that **some** steps like the peephole optimizer or the increment optimizer are on by default?

Reply:
Yeah, agree.
Rewritten it to say those are examples of steps enabled by default.